idp/alipay.go

```diff
@@ -204,17 +207,23 @@ func (idp *AlipayIdProvider) postWithBody(body interface{}, targetUrl string) ([
 	if err != nil {
 		return nil, err
 	}
-	data, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return nil, err
-	}
 	defer func(Body io.ReadCloser) {
 		err := Body.Close()
 		if err != nil {
 			return
 		}
 	}(resp.Body)
 
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.New("http status code is not 200")
+	}
+
+	utf8Reader := transform.NewReader(resp.Body,
+		simplifiedchinese.GBK.NewDecoder())
```

Comment:
Do you have to import a 3rd-party lib to do this?

Reply:
> Do you have to import a 3rd-party lib to do this?


I'll try to implement it myself

Reply:
what will happen if  the nickName of alipay user is other language like japanese、korean ...

Reply:
and it seems that we have set chatset to be utf-8 in 
https://github.com/casdoor/casdoor/blob/e99b56620be604a1a1dbe3d455a893d4f27d3040/idp/alipay.go#L102

Reply:
> what will happen if the nickName of alipay user is other language like japanese、korean ...

all use gbk, after conversion, all  language  is normal.

Reply:
> Do you have to import a 3rd-party lib to do this?

I think 3rd-party lib is more mature and robust

Reply:
> > what will happen if the nickName of alipay user is other language like japanese、korean ...
> 
> all use gbk, after conversion, all language is normal.

can we fix bug by set line 102 from `utf-8` to `UTF-8` ?

Reply:
> > > what will happen if the nickName of alipay user is other language like japanese、korean ...
> > 
> > 
> > all use gbk, after conversion, all language is normal.
> 
> can we fix bug by set line 102 from `utf-8` to `UTF-8` ?

yes

Reply:
> and it seems that we have set charset to be utf-8 in
> 
> https://github.com/casdoor/casdoor/blob/e99b56620be604a1a1dbe3d455a893d4f27d3040/idp/alipay.go#L102

encoding format for  request